tests/unit/components/BookingForm.spec.js

```diff
@@ -32,39 +32,33 @@ describe("Component BookingForm.vue", () => {
     expect(button.text()).toBe("Book");
   });
 
-  // describe('filling and submitting the "form"', () => {
-  //   let actions;
-  //   let store;
-  //
-  //   beforeEach(() => {
-  //     actions = {
-  //       addBooking: jest.fn(),
-  //     };
-  //     store = new Vuex.Store({
-  //       actions
-  //     });
-  //     underTest = shallowMount(BookingForm, { store, localVue })
-  //
-  //   });
-  //
-  //   it("should submit values from the inputs", () => {
-  //     const officeIDInput = underTest.find("#officeID");
-  //     officeIDInput.element.value = 666;
-  //     officeIDInput.trigger("change");
-  //
-  //     const emailInput = underTest.find("#email");
-  //     emailInput.element.value = "email";
-  //     emailInput.trigger("change");
-  //
-  //     const dateInput = underTest.find("#bookingDate");
-  //     dateInput.element.value = "2020-11-04";
-  //     dateInput.trigger("change");
-  //
-  //     const button = underTest.find('#btnBook');
-  //
-  //     underTest.find("#btnBook").trigger("click");
-  //
-  //     expect(actions.addBooking).toHaveBeenCalled();
-  //   });
-  // });
+  describe('filling and submitting the "form"', () => {
+    let actions;
+    let store;
+
+    beforeEach(() => {
+      actions = {
+        book: jest.fn()
+      };
+      store = new Vuex.Store({
+        actions
+      });
+      underTest = shallowMount(BookingForm, { store, localVue });
+    });
+
+    it("should submit values from the inputs", async () => {
+      const textFields = underTest.findAllComponents({
+        name: "bad-text-input"
+      });
+      const button = underTest.findComponent({ name: "bad-contained-button" });
+
+      textFields.at(0).vm.$emit("input", "Montreal");
```

Comment:
What is `.vm`

Reply:
That is reference to `Vue` instance that is bound to your components.

https://vuejs.org/v2/guide/instance.html
https://vue-test-utils.vuejs.org/api/wrapper/#trigger
https://vue-test-utils.vuejs.org/api/wrapper/#emitted

It should be possible to use `trigger` on native components but on custom component only $emit worked for me.